internal/pkg/db/redis/event.go

```diff
@@ -60,31 +60,34 @@ func unmarshalEvents(objects [][]byte, events []contract.Event) (err error) {
 	return nil
 }
 
-func unmarshalEvent(o []byte) (event contract.Event, err error) {
+func unmarshalEvent(o []byte) (contract.Event, error) {
 	var s redisEvent
 
-	err = json.Unmarshal(o, &s)
+	err := json.Unmarshal(o, &s)
 	if err != nil {
 		return contract.Event{}, err
 	}
 
-	event.ID = s.ID
-	event.Pushed = s.Pushed
-	event.Device = s.Device
-	event.Created = s.Created
-	event.Modified = s.Modified
-	event.Origin = s.Origin
+	event := contract.Event{
+		ID:       s.ID,
+		Pushed:   s.Pushed,
+		Device:   s.Device,
+		Created:  s.Created,
+		Modified: s.Modified,
+		Origin:   s.Origin,
+	}
 
 	conn, err := getConnection()
 	if err != nil {
-		return event, err
+		return contract.Event{}, err
 	}
+
```

Comment:
nitpicky, but remove blank line.  Discourages separation of the connection creation and deferred close in the future.